### PR TITLE
pAIs cannot receive Hivenet

### PR DESCRIPTION
**Pull request by Sniblet**
See title.
They inherit the ability to see all hivenet transmissions by being a silicon subtype. They should not do that. They're a consumer product, not lawbound and not trusted.

Contrary to the branch name, they aren't in any way able to receive Lii'draic.

#12635

### CODE REVIEW

**Comment by Sniblet** [comment]
!review

**Review by Geevies — CHANGES_REQUESTED** [review]
i think maybe instead of stacking istypes, there should be an overrideable on silicon that's like "can_hear_hivenet".

may be worth speaking with the vaurca team to see if we want silicons to hear hivenet in the first place, could be worth just removing entirely if they're interested, but you don't have to do that

**Comment by Sniblet** [comment]
> i think maybe instead of stacking istypes, there should be an overrideable on silicon that's like "can_hear_hivenet".

I don't know how to do this! How do I make an if statement in mob/language/station.dm read a variable in mob/living/silicon.dm?

> may be worth speaking with the vaurca team to see if we want silicons to hear hivenet in the first place, could be worth just removing entirely if they're interested, but you don't have to do that

There's lore backup for shipbounds hearing it, it looks pretty intentional.

**Comment by Sniblet** [comment]
> i think maybe instead of stacking istypes, there should be an overrideable on silicon that's like "can_hear_hivenet".
Addressed the best I knew how.

File: code/modules/mob/language/station.dm [hunk 1]
```diff
@@ -289,7 +289,9 @@
 
 /datum/language/bug/check_special_condition(var/mob/other)
 	if(istype(other, /mob/living/silicon))
-		return 1
+		var/mob/living/silicon/S = other
+		if(S.can_hear_hivenet)
+			return 1
```
Comment (SleepyGemmy) [inline code/modules/mob/language/station.dm:294]:
```suggestion
			return TRUE
```

File: code/modules/mob/living/silicon/silicon.dm [hunk 2]
```diff
@@ -49,6 +49,7 @@
 									ACCENT_SILVERSUN_EXPATRIATE, ACCENT_KONYAN, ACCENT_EARTH, ACCENT_PERSEPOLIS, ACCENT_MEDINA, ACCENT_AEMAQ, ACCENT_NEWSUEZ, ACCENT_DAMASCUS, ACCENT_ERIDANI, ACCENT_PHONG,
 									ACCENT_VISEGRAD, ACCENT_HIMEO, ACCENT_PLUTO, ACCENT_XANU)
 
+	var/can_hear_hivenet = TRUE
```
Comment (SleepyGemmy) [inline code/modules/mob/living/silicon/silicon.dm:52]:
```suggestion
	var/can_hear_hivenet = TRUE

```

File: code/modules/mob/language/station.dm (hunk 1 above)
Comment (Sniblet) [reply code/modules/mob/language/station.dm:294]: Nothing else in this section of code uses TRUE/FALSE. I'll do it anyway, but it'll look weird I think.

**Review by Geevies — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**